jaxrs2/src/main/java/io/opentracing/contrib/jaxrs/server/CurrentSpan.java

```diff
@@ -0,0 +1,22 @@
+package io.opentracing.contrib.jaxrs.server;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.core.Context;
+
+import io.opentracing.Span;
+import io.opentracing.contrib.jaxrs.SpanWrapper;
+
+
+/**
+ * @author Pavol Loffay
+ */
+public class CurrentSpan {
+
+    @Context
+    private HttpServletRequest request;
+
+    public Span injectedSpan() {
+        SpanWrapper spanWrapper = (SpanWrapper) request.getAttribute(SpanServerRequestFilter.SPAN_PROP_ID);
```

Comment:
You have several of those `#getAttribute` casting to other value in your code. Instead of doing unsafe casts all around the code, I'd suggest creating an utility class that would accept the type you want to receive the value as, logging an exception and returning null if the type is not of the expected value. As such, the target application can still be executed, even if there's a configuration issue.